cdf-core/cdf/js-modules/components/filter/FilterComponent.js

```diff
@@ -182,7 +182,7 @@ define([
             /*
              * Handle empty datasets
              */
-            if (this.query.getOption('pageSize') === 0) {
+            if (!this.query || (!configuration.component.search.serverSide && this.query.getOption('pageSize') === 0)) {
```

Comment:
the argument of this if is very long. break the components into variables on their own:
var hasQuery = !! this.query;
var isPaginated = ....